class: Workflow
cwlVersion: v1.0
id: qc_generator
label: qc_generator
$namespaces:
  s: 'https://schema.org/'
  sbg: 'https://www.sevenbridges.com/'
inputs:
  - id: reference
    type: File
    secondaryFiles:
      - ^.fasta.fai
      - ^.dict
  - id: duplex_bam
    type: File
    label: duplex_bam
    secondaryFiles:
      - ^.bai
  - id: collapsed_bam
    type: File
    label: collapsed_bam
    secondaryFiles:
      - ^.bai
  - id: group_reads_by_umi_bam
    type: File
    label: group_reads_by_umi_bam
    doc: Input BAM file generated by GroupReadByUmi.
  - id: uncollapsed_bam_base_recal
    type: File
    label: uncollapsed_bam_base_recal
    doc: An aligned SAM or BAM file.  Required.
    secondaryFiles:
      - ^.bai
  - id: pool_b_target_intervals
    type: File
    label: pool_b_target_intervals
  - id: pool_b_bait_intervals
    type: File
    label: pool_b_bait_intervals
  - id: pool_a_target_intervals
    type: File
    label: pool_a_target_intervals
  - id: pool_a_bait_intervals
    type: File
    label: pool_a_bait_intervals
  - id: noise_sites_bed
    type: File
    label: noise_sites_bed
    doc: >-
      Path to BED file containing regions over which to calculate noise
      [required]
  - id: sample_sex
    type: string?
    doc: Expected sample sex (i.e. M or F).
  - id: sample_name
    type: string?
    doc: >-
      Sample name. If not specified, sample name is automatically figured out
      from the BAM file.
  - id: sample_group
    type: string
    doc: The sample group (e.g. the sample patient ID).
  - id: simplex_bam
    type: File
    label: simplex_bam
    secondaryFiles:
      - ^.bai
  - id: biometrics_plot
    type: boolean?
    label: biometrics_plot
    doc: Also output plots of the data.
  - id: duplex_biometrics_major_threshold
    type: float?
    label: duplex_biometrics_major_threshold
    doc: Major contamination threshold for bad sample.
  - id: biometrics_json
    type: boolean?
    label: biometrics_json
    doc: Also output data in JSON format.
  - id: collapsed_biometrics_minor_threshold
    type: float?
    label: collapsed_biometrics_minor_threshold
    doc: Minor contamination threshold for bad sample.
  - id: collapsed_biometrics_coverage_threshold
    type: int?
    label: collapsed_biometrics_coverage_threshold
    doc: Samples with Y chromosome above this value will be considered male.
  - id: sequence_qc_min_basq
    type: int?
  - id: sequence_qc_min_mapq
    type: int?
  - id: sequence_qc_threshold
    type: float?
  - id: sequence_qc_truncate
    type: int?
  - id: hsmetrics_minimum_mapping_quality
    type: int?
  - id: hsmetrics_minimum_base_quality
    type: int?
  - id: hsmetrics_coverage_cap
    type: int?
  - id: vcf_file
    type: File
  - id: hotspots_maf
    type: File
outputs:
  - id: uncollapsed_bam_stats_pool_a_dir
    outputSource:
      - uncollapsed_bam_stats_pool_a/directory
    type: Directory
    label: uncollapsed_bam_stats_pool_a_dir
  - id: uncollapsed_bam_stats_pool_b_dir
    outputSource:
      - uncollapsed_bam_stats_pool_b/directory
    type: Directory
    label: uncollapsed_bam_stats_pool_b_dir
  - id: gatk_mean_quality_by_cycle_recal_dir
    outputSource:
      - gatk_mean_quality_by_cycle_recal/directory
    type: Directory
    label: gatk_mean_quality_by_cycle_recal_dir
  - id: collapsed_bam_biometrics_dir
    outputSource:
      - collapsed_bam_biometrics/directory
    type: Directory
    label: collapsed_bam_biometrics_dir
  - id: collapsed_bam_duplex_metrics_pool_b_dir
    outputSource:
      - collapsed_bam_duplex_metrics_pool_b/directory
    type: Directory
    label: collapsed_bam_duplex_metrics_pool_b_dir
  - id: collapsed_bam_duplex_metrics_pool_a_dir
    outputSource:
      - collapsed_bam_duplex_metrics_pool_a/directory
    type: Directory
    label: collapsed_bam_duplex_metrics_pool_a_dir
  - id: collapsed_bam_stats_pool_b_dir
    outputSource:
      - collapsed_bam_stats_pool_b/directory
    type: Directory
    label: collapsed_bam_stats_pool_b_dir
  - id: collapsed_bam_stats_pool_a_dir
    outputSource:
      - collapsed_bam_stats_pool_a/directory
    type: Directory
    label: collapsed_bam_stats_pool_a_dir
  - id: simplex_bam_pool_a_dir
    outputSource:
      - simplex_bam_pool_a/directory
    type: Directory
    label: simplex_bam_pool_a_dir
  - id: simplex_bam_pool_b_dir
    outputSource:
      - simplex_bam_pool_b/directory
    type: Directory
    label: simplex_bam_pool_b_dir
  - id: duplex_bam_sequence_qc_dir
    outputSource:
      - duplex_bam_sequence_qc/directory
    type: Directory
    label: duplex_bam_sequence_qc_dir
  - id: duplex_bam_stats_pool_a_dir
    outputSource:
      - duplex_bam_stats_pool_a/directory
    type: Directory
    label: duplex_bam_stats_pool_a_dir
  - id: duplex_bam_stats_pool_b_dir
    outputSource:
      - duplex_bam_stats_pool_b/directory
    type: Directory
    label: duplex_bam_stats_pool_b_dir
  - id: duplex_bam_biometrics_dir
    outputSource:
      - duplex_bam_biometrics/directory
    type: Directory
    label: duplex_bam_biometrics_dir
  - id: duplex_bam_genotyping_maf
    outputSource:
      - qc_duplex_bam/fillout_maf
    type: File
    label: duplex_bam_genotyping_maf
  - id: duplex_biometrics_extract_pickle
    outputSource:
      - qc_duplex_bam/biometrics_extract_pickle
    type: File
  - id: collapsed_biometrics_extract_pickle
    outputSource:
      - qc_collapsed_bam/biometrics_extract_pickle
    type: File
  - id: collapsed_bam_genotyping_maf
    outputSource:
      - qc_collapsed_bam/fillout_maf
    type: File
steps:
  - id: qc_collapsed_bam
    in:
      - id: reference
        source: reference
      - id: pool_b_target_intervals
        source: pool_b_target_intervals
      - id: pool_a_target_intervals
        source: pool_a_target_intervals
      - id: collapsed_bam
        source:
          - collapsed_bam
      - id: group_reads_by_umi_bam
        source:
          - group_reads_by_umi_bam
      - id: pool_a_bait_intervals
        source: pool_a_bait_intervals
      - id: pool_b_bait_intervals
        source: pool_b_bait_intervals
      - id: json
        source: biometrics_json
      - id: plot
        source: biometrics_plot
      - id: minor_threshold
        source: collapsed_biometrics_minor_threshold
      - id: coverage_threshold
        source: collapsed_biometrics_coverage_threshold
      - id: hsmetrics_minimum_mapping_quality
        source: hsmetrics_minimum_mapping_quality
      - id: hsmetrics_minimum_base_quality
        source: hsmetrics_minimum_base_quality
      - id: hsmetrics_coverage_cap
        source: hsmetrics_coverage_cap
      - id: vcf_file
        source: vcf_file
      - id: sample_name
        source: sample_name
      - id: sample_sex
        source: sample_sex
      - id: sample_group
        source: sample_group
      - id: maf
        source: hotspots_maf
    out:
      - id: fgbio_collect_duplex_seq_metrics_duplex_family_size_pool_a
      - id: fgbio_collect_duplex_seq_metrics_duplex_qc_pool_a
      - id: fgbio_collect_duplex_seq_metrics_duplex_pool_a
      - id: fgbio_collect_duplex_seq_metrics_duplex_yield_metrics_pool_a
      - id: fgbio_collect_duplex_seq_metrics_family_size_pool_a
      - id: fgbio_collect_duplex_seq_metrics_umi_counts_pool_a
      - id: fgbio_collect_duplex_seq_metrics_duplex_family_size_pool_b
      - id: fgbio_collect_duplex_seq_metrics_duplex_qc_pool_b
      - id: fgbio_collect_duplex_seq_metrics_duplex_umi_counts_pool_b
      - id: fgbio_collect_duplex_seq_metrics_duplex_yield_metrics_pool_b
      - id: fgbio_collect_duplex_seq_metrics_family_size_pool_b
      - id: fgbio_collect_duplex_seq_metrics_umi_counts_pool_b
      - id: biometrics_minor_csv
      - id: biometrics_minor_json
      - id: biometrics_minor_plot
      - id: biometrics_minor_sites_plot
      - id: biometrics_sexmismatch_json
      - id: biometrics_sexmismatch_csv
      - id: gatk_collect_insert_size_metrics_txt_pool_b
      - id: gatk_collect_insert_size_metrics_histogram_pdf_pool_b
      - id: gatk_collect_hs_metrics_txt_pool_b
      - id: gatk_collect_hs_metrics_per_target_coverage_txt_pool_b
      - id: gatk_collect_hs_metrics_per_base_coverage_txt_pool_b
      - id: gatk_collect_alignment_summary_metrics_txt_pool_b
      - id: gatk_collect_insert_size_metrics_txt_pool_a
      - id: gatk_collect_insert_size_metrics_histogram_pdf_pool_a
      - id: gatk_collect_hs_metrics_txt_pool_a
      - id: gatk_collect_hs_metrics_per_target_coverage_txt_pool_a
      - id: gatk_collect_hs_metrics_per_base_coverage_txt_pool_a
      - id: gatk_collect_alignment_summary_metrics_txt_pool_a
      - id: biometrics_major_plot
      - id: biometrics_major_json
      - id: biometrics_major_csv
      - id: biometrics_extract_pickle
      - id: fillout_maf
    run: ../cwl_subworkflows/qc_collapsed_bam/qc_collapsed_bam.cwl
    label: qc_collapsed_bam
  - id: qc_uncollapsed_bam
    in:
      - id: reference
        source: reference
      - id: uncollapsed_bam_base_recal
        source:
          - uncollapsed_bam_base_recal
      - id: pool_b_target_intervals
        source: pool_b_target_intervals
      - id: pool_b_bait_intervals
        source: pool_b_bait_intervals
      - id: pool_a_bait_intervals
        source: pool_a_bait_intervals
      - id: pool_a_target_intervals
        source: pool_a_target_intervals
      - id: hsmetrics_minimum_mapping_quality
        source: hsmetrics_minimum_mapping_quality
      - id: hsmetrics_minimum_base_quality
        source: hsmetrics_minimum_base_quality
      - id: hsmetrics_coverage_cap
        source: hsmetrics_coverage_cap
    out:
      - id: gatk_collect_alignment_summary_metrics_txt_pool_b
      - id: gatk_collect_hs_metrics_per_base_coverage_txt_pool_b
      - id: gatk_collect_hs_metrics_per_target_coverage_txt_pool_b
      - id: gatk_collect_hs_metrics_txt_pool_b
      - id: gatk_collect_insert_size_metrics_histogram_pdf_pool_b
      - id: gatk_collect_insert_size_metrics_txt_pool_b
      - id: gatk_collect_alignment_summary_metrics_txt_pool_a
      - id: gatk_collect_hs_metrics_per_base_coverage_txt_pool_a
      - id: gatk_collect_hs_metrics_per_target_coverage_txt_pool_a
      - id: gatk_collect_hs_metrics_txt_pool_a
      - id: gatk_collect_insert_size_metrics_histogram_pdf_pool_a
      - id: gatk_collect_insert_size_metrics_txt_pool_a
      - id: gatk_mean_quality_by_cycle_output_base_recal
      - id: gatk_mean_quality_by_cycle_chart_output_base_recal
    run: ../cwl_subworkflows/qc_uncollapsed_bam/qc_uncollapsed_bam.cwl
    label: qc_uncollapsed_bam
  - id: qc_duplex_bam
    in:
      - id: reference
        source: reference
      - id: duplex_bam
        source:
          - duplex_bam
      - id: pool_a_target_intervals
        source: pool_a_target_intervals
      - id: pool_a_bait_intervals
        source: pool_a_bait_intervals
      - id: pool_b_target_intervals
        source: pool_b_target_intervals
      - id: pool_b_bait_intervals
        source: pool_b_bait_intervals
      - id: noise_sites_bed
        source: noise_sites_bed
      - id: sample_name
        source:
          - sample_name
      - id: plot
        source: biometrics_plot
      - id: json
        source: biometrics_json
      - id: sequence_qc_min_basq
        source: sequence_qc_min_basq
      - id: sequence_qc_min_mapq
        source: sequence_qc_min_mapq
      - id: sequence_qc_threshold
        source: sequence_qc_threshold
      - id: sequence_qc_truncate
        source: sequence_qc_truncate
      - id: hsmetrics_minimum_mapping_quality
        source: hsmetrics_minimum_mapping_quality
      - id: hsmetrics_minimum_base_quality
        source: hsmetrics_minimum_base_quality
      - id: hsmetrics_coverage_cap
        source: hsmetrics_coverage_cap
      - id: major_threshold
        source: duplex_biometrics_major_threshold
      - id: vcf_file
        source: vcf_file
      - id: sample_sex
        source: sample_sex
      - id: sample_group
        source: sample_group
      - id: maf
        source: hotspots_maf
    out:
      - id: sequence_qc_noise_positions
      - id: sequence_qc_noise_n
      - id: sequence_qc_noise_del
      - id: sequence_qc_noise_acgt
      - id: sequence_qc_figures
      - id: gatk_collect_alignment_summary_metrics_txt_pool_b
      - id: gatk_collect_hs_metrics_per_base_coverage_txt_pool_b
      - id: gatk_collect_hs_metrics_per_target_coverage_txt_pool_b
      - id: gatk_collect_hs_metrics_txt_pool_b
      - id: gatk_collect_insert_size_metrics_histogram_pdf_pool_b
      - id: gatk_collect_insert_size_metrics_txt_pool_b
      - id: gatk_collect_alignment_summary_metrics_txt_pool_a
      - id: gatk_collect_hs_metrics_per_base_coverage_txt_pool_a
      - id: gatk_collect_hs_metrics_per_target_coverage_txt_pool_a
      - id: gatk_collect_hs_metrics_txt_pool_a
      - id: gatk_collect_insert_size_metrics_histogram_pdf_pool_a
      - id: gatk_collect_insert_size_metrics_txt_pool_a
      - id: sequence_qc_pileup
      - id: sequence_qc_noise_by_substitution
      - id: biometrics_major_plot
      - id: biometrics_major_json
      - id: biometrics_major_csv
      - id: biometrics_extract_pickle
      - id: biometrics_minor_sites_plot
      - id: biometrics_minor_plot
      - id: biometrics_minor_json
      - id: biometrics_minor_csv
      - id: fillout_maf
    run: ../cwl_subworkflows/qc_duplex_bam/qc_duplex_bam.cwl
    label: qc_duplex_bam
  - id: simplex_bam_pool_a
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_simplex_bam/gatk_collect_insert_size_metrics_txt_pool_a
          - qc_simplex_bam/gatk_collect_insert_size_metrics_histogram_pdf_pool_a
          - qc_simplex_bam/gatk_collect_hs_metrics_txt_pool_a
          - >-
            qc_simplex_bam/gatk_collect_hs_metrics_per_target_coverage_txt_pool_a
          - qc_simplex_bam/gatk_collect_hs_metrics_per_base_coverage_txt_pool_a
          - qc_simplex_bam/gatk_collect_alignment_summary_metrics_txt_pool_a
      - id: output_directory_name
        default: simplex_bam_pool_a
        source: sample_name
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: simplex_bam_pool_a
  - id: simplex_bam_pool_b
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_simplex_bam/gatk_collect_insert_size_metrics_txt_pool_b
          - qc_simplex_bam/gatk_collect_insert_size_metrics_histogram_pdf_pool_b
          - qc_simplex_bam/gatk_collect_hs_metrics_txt_pool_b
          - >-
            qc_simplex_bam/gatk_collect_hs_metrics_per_target_coverage_txt_pool_b
          - qc_simplex_bam/gatk_collect_hs_metrics_per_base_coverage_txt_pool_b
          - qc_simplex_bam/gatk_collect_alignment_summary_metrics_txt_pool_b
      - id: output_directory_name
        default: simplex_bam_pool_b
        source: sample_name
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: simplex_bam_pool_b
  - id: uncollapsed_bam_stats_pool_b
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_uncollapsed_bam/gatk_collect_alignment_summary_metrics_txt_pool_b
          - >-
            qc_uncollapsed_bam/gatk_collect_hs_metrics_per_base_coverage_txt_pool_b
          - >-
            qc_uncollapsed_bam/gatk_collect_hs_metrics_per_target_coverage_txt_pool_b
          - qc_uncollapsed_bam/gatk_collect_hs_metrics_txt_pool_b
          - >-
            qc_uncollapsed_bam/gatk_collect_insert_size_metrics_histogram_pdf_pool_b
          - qc_uncollapsed_bam/gatk_collect_insert_size_metrics_txt_pool_b
      - id: output_directory_name
        default: uncollapsed_bam_stats_pool_b
        source: sample_name
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: uncollapsed_bam_stats_pool_b
  - id: uncollapsed_bam_stats_pool_a
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_uncollapsed_bam/gatk_collect_alignment_summary_metrics_txt_pool_a
          - >-
            qc_uncollapsed_bam/gatk_collect_hs_metrics_per_base_coverage_txt_pool_a
          - >-
            qc_uncollapsed_bam/gatk_collect_hs_metrics_per_target_coverage_txt_pool_a
          - qc_uncollapsed_bam/gatk_collect_hs_metrics_txt_pool_a
          - >-
            qc_uncollapsed_bam/gatk_collect_insert_size_metrics_histogram_pdf_pool_a
          - qc_uncollapsed_bam/gatk_collect_insert_size_metrics_txt_pool_a
      - id: output_directory_name
        default: uncollapsed_bam_stats_pool_a
        source: sample_name
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: uncollapsed_bam_stats_pool_a
  - id: gatk_mean_quality_by_cycle_recal
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - >-
            qc_uncollapsed_bam/gatk_mean_quality_by_cycle_chart_output_base_recal
          - qc_uncollapsed_bam/gatk_mean_quality_by_cycle_output_base_recal
      - id: output_directory_name
        default: gatk_mean_quality_by_cycle_recal
        source: sample_name
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: gatk_mean_quality_by_cycle_recal
  - id: collapsed_bam_stats_pool_a
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_collapsed_bam/gatk_collect_insert_size_metrics_txt_pool_a
          - >-
            qc_collapsed_bam/gatk_collect_insert_size_metrics_histogram_pdf_pool_a
          - qc_collapsed_bam/gatk_collect_hs_metrics_txt_pool_a
          - >-
            qc_collapsed_bam/gatk_collect_hs_metrics_per_target_coverage_txt_pool_a
          - >-
            qc_collapsed_bam/gatk_collect_hs_metrics_per_base_coverage_txt_pool_a
          - qc_collapsed_bam/gatk_collect_alignment_summary_metrics_txt_pool_a
      - id: output_directory_name
        default: collapsed_bam_stats_pool_a
        source: sample_name
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: collapsed_bam_stats_pool_a
  - id: collapsed_bam_stats_pool_b
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_collapsed_bam/gatk_collect_insert_size_metrics_txt_pool_b
          - >-
            qc_collapsed_bam/gatk_collect_insert_size_metrics_histogram_pdf_pool_b
          - qc_collapsed_bam/gatk_collect_hs_metrics_txt_pool_b
          - >-
            qc_collapsed_bam/gatk_collect_hs_metrics_per_target_coverage_txt_pool_b
          - >-
            qc_collapsed_bam/gatk_collect_hs_metrics_per_base_coverage_txt_pool_b
          - qc_collapsed_bam/gatk_collect_alignment_summary_metrics_txt_pool_b
      - id: output_directory_name
        default: collapsed_bam_stats_pool_b
        source: sample_name
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: collapsed_bam_stats_pool_b
  - id: collapsed_bam_duplex_metrics_pool_a
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_collapsed_bam/fgbio_collect_duplex_seq_metrics_umi_counts_pool_a
          - qc_collapsed_bam/fgbio_collect_duplex_seq_metrics_family_size_pool_a
          - >-
            qc_collapsed_bam/fgbio_collect_duplex_seq_metrics_duplex_yield_metrics_pool_a
          - qc_collapsed_bam/fgbio_collect_duplex_seq_metrics_duplex_qc_pool_a
          - qc_collapsed_bam/fgbio_collect_duplex_seq_metrics_duplex_pool_a
          - >-
            qc_collapsed_bam/fgbio_collect_duplex_seq_metrics_duplex_family_size_pool_a
      - id: output_directory_name
        default: collapsed_bam_duplex_metrics_pool_a
        source: sample_name
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: collapsed_bam_duplex_metrics_pool_a
  - id: collapsed_bam_duplex_metrics_pool_b
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_collapsed_bam/fgbio_collect_duplex_seq_metrics_umi_counts_pool_b
          - qc_collapsed_bam/fgbio_collect_duplex_seq_metrics_family_size_pool_b
          - >-
            qc_collapsed_bam/fgbio_collect_duplex_seq_metrics_duplex_yield_metrics_pool_b
          - >-
            qc_collapsed_bam/fgbio_collect_duplex_seq_metrics_duplex_umi_counts_pool_b
          - qc_collapsed_bam/fgbio_collect_duplex_seq_metrics_duplex_qc_pool_b
          - >-
            qc_collapsed_bam/fgbio_collect_duplex_seq_metrics_duplex_family_size_pool_b
      - id: output_directory_name
        default: collapsed_bam_duplex_metrics_pool_b
        source: sample_name
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: collapsed_bam_duplex_metrics_pool_b
  - id: collapsed_bam_biometrics
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_collapsed_bam/biometrics_sexmismatch_json
          - qc_collapsed_bam/biometrics_sexmismatch_csv
          - qc_collapsed_bam/biometrics_minor_sites_plot
          - qc_collapsed_bam/biometrics_minor_plot
          - qc_collapsed_bam/biometrics_minor_json
          - qc_collapsed_bam/biometrics_minor_csv
          - qc_collapsed_bam/biometrics_major_plot
          - qc_collapsed_bam/biometrics_major_json
          - qc_collapsed_bam/biometrics_major_csv
          - qc_collapsed_bam/biometrics_extract_pickle
          - qc_collapsed_bam/biometrics_minor_sites_plot
          - qc_collapsed_bam/biometrics_sexmismatch_csv
          - qc_collapsed_bam/biometrics_sexmismatch_json
      - id: output_directory_name
        default: collapsed_bam_biometrics
        source: sample_name
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: collapsed_bam_biometrics
  - id: duplex_bam_sequence_qc
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_duplex_bam/sequence_qc_noise_positions
          - qc_duplex_bam/sequence_qc_noise_n
          - qc_duplex_bam/sequence_qc_noise_del
          - qc_duplex_bam/sequence_qc_noise_acgt
          - qc_duplex_bam/sequence_qc_figures
          - qc_duplex_bam/sequence_qc_pileup
          - qc_duplex_bam/sequence_qc_noise_by_substitution
      - id: output_directory_name
        default: duplex_bam_sequence_qc
        source: sample_name
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: duplex_bam_sequence_qc
  - id: duplex_bam_stats_pool_a
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_duplex_bam/gatk_collect_insert_size_metrics_txt_pool_a
          - qc_duplex_bam/gatk_collect_insert_size_metrics_histogram_pdf_pool_a
          - qc_duplex_bam/gatk_collect_hs_metrics_txt_pool_a
          - qc_duplex_bam/gatk_collect_hs_metrics_per_target_coverage_txt_pool_a
          - qc_duplex_bam/gatk_collect_hs_metrics_per_base_coverage_txt_pool_a
          - qc_duplex_bam/gatk_collect_alignment_summary_metrics_txt_pool_a
      - id: output_directory_name
        default: duplex_bam_stats_pool_a
        source: sample_name
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: duplex_bam_stats_pool_a
  - id: duplex_bam_stats_pool_b
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_duplex_bam/gatk_collect_insert_size_metrics_txt_pool_b
          - qc_duplex_bam/gatk_collect_insert_size_metrics_histogram_pdf_pool_b
          - qc_duplex_bam/gatk_collect_hs_metrics_txt_pool_b
          - qc_duplex_bam/gatk_collect_hs_metrics_per_target_coverage_txt_pool_b
          - qc_duplex_bam/gatk_collect_hs_metrics_per_base_coverage_txt_pool_b
          - qc_duplex_bam/gatk_collect_alignment_summary_metrics_txt_pool_b
      - id: output_directory_name
        default: duplex_bam_stats_pool_b
        source: sample_name
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: duplex_bam_stats_pool_b
  - id: duplex_bam_biometrics
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_duplex_bam/biometrics_major_csv
          - qc_duplex_bam/biometrics_major_json
          - qc_duplex_bam/biometrics_major_plot
          - qc_duplex_bam/biometrics_minor_csv
          - qc_duplex_bam/biometrics_minor_json
          - qc_duplex_bam/biometrics_minor_plot
          - qc_duplex_bam/biometrics_minor_sites_plot
          - qc_duplex_bam/biometrics_extract_pickle
      - id: output_directory_name
        default: duplex_bam_biometrics
        source: sample_name
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: duplex_bam_biometrics
  - id: qc_simplex_bam
    in:
      - id: reference
        source: reference
      - id: simplex_bam
        source: simplex_bam
      - id: pool_b_target_intervals
        source: pool_b_target_intervals
      - id: pool_b_bait_intervals
        source: pool_b_bait_intervals
      - id: pool_a_bait_intervals
        source: pool_a_bait_intervals
      - id: pool_a_target_intervals
        source: pool_a_target_intervals
      - id: hsmetrics_minimum_mapping_quality
        source: hsmetrics_minimum_mapping_quality
      - id: hsmetrics_minimum_base_quality
        source: hsmetrics_minimum_base_quality
      - id: hsmetrics_coverage_cap
        source: hsmetrics_coverage_cap
    out:
      - id: gatk_collect_alignment_summary_metrics_txt_pool_b
      - id: gatk_collect_hs_metrics_per_base_coverage_txt_pool_b
      - id: gatk_collect_hs_metrics_per_target_coverage_txt_pool_b
      - id: gatk_collect_hs_metrics_txt_pool_b
      - id: gatk_collect_insert_size_metrics_histogram_pdf_pool_b
      - id: gatk_collect_insert_size_metrics_txt_pool_b
      - id: gatk_collect_alignment_summary_metrics_txt_pool_a
      - id: gatk_collect_hs_metrics_per_base_coverage_txt_pool_a
      - id: gatk_collect_hs_metrics_per_target_coverage_txt_pool_a
      - id: gatk_collect_hs_metrics_txt_pool_a
      - id: gatk_collect_insert_size_metrics_histogram_pdf_pool_a
      - id: gatk_collect_insert_size_metrics_txt_pool_a
    run: ../cwl_subworkflows/qc_simplex_bam/qc_simplex_bam.cwl
    label: qc_simplex_bam
requirements:
  - class: SubworkflowFeatureRequirement
  - class: MultipleInputFeatureRequirement
  - class: StepInputExpressionRequirement
$schemas:
  - 'http://schema.org/version/latest/schemaorg-current-http.rdf'
's:author':
  - class: 's:Person'
    's:email': 'mailto:murphyc4@mskcc.org'
    's:identifier': ''
    's:name': Charlie Murphy
's:citation': ''
's:codeRepository': 'https://github.com/msk-access/cwl_subworkflows'
's:contributor':
  - class: 's:Person'
    's:email': 'mailto:murphyc4@mskcc.org'
    's:name': Charlie Murphy
's:dateCreated': '2021-05-19'
's:license': 'https://spdx.org/licenses/Apache-2.0'
