class: Workflow
cwlVersion: v1.0
id: qc_generator
label: qc_generator
$namespaces:
  s: 'https://schema.org/'
  sbg: 'https://www.sevenbridges.com/'
inputs:
  - id: reference
    type: File
    secondaryFiles:
      - ^.fasta.fai
      - ^.dict
    'sbg:x': -1367.075439453125
    'sbg:y': -170.63369750976562
  - id: duplex_bam
    type: File
    label: duplex_bam
    secondaryFiles:
      - ^.bai
    'sbg:x': -808.21337890625
    'sbg:y': -258.8006591796875
  - id: collapsed_bam
    type: File
    label: collapsed_bam
    secondaryFiles:
      - ^.bai
    'sbg:x': -810.0906372070312
    'sbg:y': -123.79762268066406
  - id: group_reads_by_umi_bam
    type: File
    label: group_reads_by_umi_bam
    doc: Input BAM file generated by GroupReadByUmi.
    'sbg:x': -811.8580322265625
    'sbg:y': 27.081497192382812
  - id: uncollapsed_bam
    type: File
    label: uncollapsed_bam
    secondaryFiles:
      - ^.bai
    'sbg:x': -810.2417602539062
    'sbg:y': 164.8699493408203
  - id: uncollapsed_bam_base_recal
    type: File
    label: uncollapsed_bam_base_recal
    doc: An aligned SAM or BAM file.  Required.
    secondaryFiles:
      - ^.bai
    'sbg:x': -813.2417602539062
    'sbg:y': 310.27471923828125
  - id: pool_b_target_intervals
    type: File
    label: pool_b_target_intervals
    'sbg:x': -1362.9149169921875
    'sbg:y': 304.2558288574219
  - id: pool_b_bait_intervals
    type: File
    label: pool_b_bait_intervals
    'sbg:x': -1362.1119384765625
    'sbg:y': 466.5614929199219
  - id: pool_a_target_intervals
    type: File
    label: pool_a_target_intervals
    'sbg:x': -1358.999755859375
    'sbg:y': -23.60011863708496
  - id: pool_a_bait_intervals
    type: File
    label: pool_a_bait_intervals
    'sbg:x': -1366.56494140625
    'sbg:y': 126.96497344970703
  - id: biometrics_bed_file
    type: File?
    doc: BED file containing the intervals to be queried.
    'sbg:x': -1356.0794677734375
    'sbg:y': 652.2377319335938
  - id: biometrics_vcf_file
    type: File
    doc: VCF file containing the SNPs to be queried.
    'sbg:x': -1345.8929443359375
    'sbg:y': 834.68603515625
  - id: noise_sites_bed
    type: File
    label: noise_sites_bed
    doc: >-
      Path to BED file containing regions over which to calculate noise
      [required]
    'sbg:x': -1351.35888671875
    'sbg:y': 982.7118530273438
  - id: sample_sex
    type: string?
    doc: Expected sample sex (i.e. M or F).
    'sbg:x': -811.90576171875
    'sbg:y': -766.7771606445312
  - id: sample_name
    type: string?
    doc: >-
      Sample name. If not specified, sample name is automatically figured out
      from the BAM file.
    'sbg:x': -807.0020751953125
    'sbg:y': -645.0062255859375
  - id: sample_group
    type: string
    doc: The sample group (e.g. the sample patient ID).
    'sbg:x': -813.4239501953125
    'sbg:y': -524.909912109375
  - id: simplex_bam
    type: File
    label: simplex_bam
    secondaryFiles:
      - ^.bai
    'sbg:x': -809.8264770507812
    'sbg:y': -383.9412536621094
  - id: biometrics_plot
    type: boolean?
    label: biometrics_plot
    doc: Also output plots of the data.
    'sbg:x': -1379.9268798828125
    'sbg:y': -906.622314453125
  - id: duplex_biometrics_minor_threshold
    type: float?
    label: duplex_biometrics_minor_threshold
    doc: Minor contamination threshold for bad sample.
    'sbg:x': -1403.07958984375
    'sbg:y': -1674.0557861328125
  - id: duplex_biometrics_min_base_quality
    type: int?
    label: duplex_biometrics_min_base_quality
    doc: Minimum base quality of reads to be used for pileup.
    'sbg:x': -1384.262451171875
    'sbg:y': -1172.6351318359375
  - id: duplex_biometrics_major_threshold
    type: float?
    label: duplex_biometrics_major_threshold
    doc: Major contamination threshold for bad sample.
    'sbg:x': -1390.1893310546875
    'sbg:y': -1050.3974609375
  - id: biometrics_json
    type: boolean?
    label: biometrics_json
    doc: Also output data in JSON format.
    'sbg:x': -1376.9451904296875
    'sbg:y': -753.5125732421875
  - id: collapsed_biometrics_minor_threshold
    type: float?
    label: collapsed_biometrics_minor_threshold
    doc: Minor contamination threshold for bad sample.
    'sbg:x': -838.6036376953125
    'sbg:y': -2211.296142578125
  - id: collapsed_biometrics_min_mapping_quality
    type: int?
    label: collapsed_biometrics_min_mapping_quality
    doc: Minimum mapping quality of reads to be used for pileup.
    'sbg:x': -842.24755859375
    'sbg:y': -2087.296142578125
  - id: collapsed_biometrics_min_homozygous_thresh
    type: float?
    label: collapsed_biometrics_min_homozygous_thresh
    doc: Minimum threshold to define homozygous.
    'sbg:x': -838.9255981445312
    'sbg:y': -1947.6180419921875
  - id: collapsed_biometrics_min_coverage
    type: int?
    label: collapsed_biometrics_min_coverage
    doc: Minimum coverage to count a site.
    'sbg:x': -833.9255981445312
    'sbg:y': -1817.9400634765625
  - id: collapsed_biometrics_min_base_quality
    type: int?
    label: collapsed_biometrics_min_base_quality
    doc: Minimum base quality of reads to be used for pileup.
    'sbg:x': -835.4125366210938
    'sbg:y': -1676.795166015625
  - id: collapsed_biometrics_coverage_threshold
    type: int?
    label: collapsed_biometrics_coverage_threshold
    doc: Samples with Y chromosome above this value will be considered male.
    'sbg:x': -830.3240356445312
    'sbg:y': -1386.815185546875
  - id: sequence_qc_min_basq
    type: int?
    'sbg:x': -1393.9599609375
    'sbg:y': -1809.91650390625
  - id: sequence_qc_min_mapq
    type: int?
    'sbg:x': -1398.4918212890625
    'sbg:y': -1940.338134765625
  - id: sequence_qc_threshold
    type: float?
    'sbg:x': -1405.4483642578125
    'sbg:y': -2085.8505859375
  - id: sequence_qc_truncate
    type: int?
    'sbg:x': -1403.2008056640625
    'sbg:y': -2227.529296875
  - id: hsmetrics_minimum_mapping_quality
    type: int?
    'sbg:x': -1366.3743896484375
    'sbg:y': -306.4041442871094
  - id: hsmetrics_minimum_base_quality
    type: int?
    'sbg:x': -1361.5120849609375
    'sbg:y': -422.63983154296875
  - id: hsmetrics_coverage_cap
    type: int?
    'sbg:x': -1374.583984375
    'sbg:y': -534.837890625
  - id: vcf_file
    type: File
    'sbg:x': -936.71826171875
    'sbg:y': -914.0149536132812
outputs:
  - id: uncollapsed_bam_stats_pool_a_dir
    outputSource:
      - uncollapsed_bam_stats_pool_a/directory
    type: Directory
    label: uncollapsed_bam_stats_pool_a_dir
    'sbg:x': 1349.0845947265625
    'sbg:y': 1307.4302978515625
  - id: uncollapsed_bam_stats_pool_b_dir
    outputSource:
      - uncollapsed_bam_stats_pool_b/directory
    type: Directory
    label: uncollapsed_bam_stats_pool_b_dir
    'sbg:x': 1252.303466796875
    'sbg:y': 1104
  - id: gatk_mean_quality_by_cycle_dir
    outputSource:
      - gatk_mean_quality_by_cycle/directory
    type: Directory
    label: gatk_mean_quality_by_cycle_dir
    'sbg:x': 1184.462646484375
    'sbg:y': 928.5895385742188
  - id: gatk_mean_quality_by_cycle_recal_dir
    outputSource:
      - gatk_mean_quality_by_cycle_recal/directory
    type: Directory
    label: gatk_mean_quality_by_cycle_recal_dir
    'sbg:x': 1137.0323486328125
    'sbg:y': 743.6890258789062
  - id: collapsed_bam_biometrics_dir
    outputSource:
      - collapsed_bam_biometrics/directory
    type: Directory
    label: collapsed_bam_biometrics_dir
    'sbg:x': 1376.462646484375
    'sbg:y': 399
  - id: collapsed_bam_duplex_metrics_pool_b_dir
    outputSource:
      - collapsed_bam_duplex_metrics_pool_b/directory
    type: Directory
    label: collapsed_bam_duplex_metrics_pool_b_dir
    'sbg:x': 1782.5870361328125
    'sbg:y': 271
  - id: collapsed_bam_duplex_metrics_pool_a_dir
    outputSource:
      - collapsed_bam_duplex_metrics_pool_a/directory
    type: Directory
    label: collapsed_bam_duplex_metrics_pool_a_dir
    'sbg:x': 1745.2164306640625
    'sbg:y': 141
  - id: collapsed_bam_stats_pool_b_dir
    outputSource:
      - collapsed_bam_stats_pool_b/directory
    type: Directory
    label: collapsed_bam_stats_pool_b_dir
    'sbg:x': 1744.726318359375
    'sbg:y': 15
  - id: collapsed_bam_stats_pool_a_dir
    outputSource:
      - collapsed_bam_stats_pool_a/directory
    type: Directory
    label: collapsed_bam_stats_pool_a_dir
    'sbg:x': 1478.7935791015625
    'sbg:y': -114
  - id: simplex_bam_pool_a_dir
    outputSource:
      - simplex_bam_pool_a/directory
    type: Directory
    label: simplex_bam_pool_a_dir
    'sbg:x': 683
    'sbg:y': -1466.514892578125
  - id: simplex_bam_pool_b_dir
    outputSource:
      - simplex_bam_pool_b/directory
    type: Directory
    label: simplex_bam_pool_b_dir
    'sbg:x': 800.7412719726562
    'sbg:y': -1164.8134765625
  - id: duplex_bam_sequence_qc_dir
    outputSource:
      - duplex_bam_sequence_qc/directory
    type: Directory
    label: duplex_bam_sequence_qc_dir
    'sbg:x': 787.3706665039062
    'sbg:y': -865.6616821289062
  - id: duplex_bam_stats_pool_a_dir
    outputSource:
      - duplex_bam_stats_pool_a/directory
    type: Directory
    label: duplex_bam_stats_pool_a_dir
    'sbg:x': 1173.02490234375
    'sbg:y': -608.9801025390625
  - id: duplex_bam_stats_pool_b_dir
    outputSource:
      - duplex_bam_stats_pool_b/directory
    type: Directory
    label: duplex_bam_stats_pool_b_dir
    'sbg:x': 1357.6268310546875
    'sbg:y': -460.54974365234375
  - id: duplex_bam_biometrics_dir
    outputSource:
      - duplex_bam_biometrics/directory
    type: Directory
    label: duplex_bam_biometrics_dir
    'sbg:x': 1472.6318359375
    'sbg:y': -329.2865295410156
  - id: collapsed_bam_biometrics_extract_file
    outputSource:
      - qc_collapsed_bam/biometrics_extract_pickle
    type: 'File'
    'sbg:x': 111.16825103759766
    'sbg:y': 671.5779418945312
  - id: duplex_bam_biometrics_extract_file
    outputSource:
      - qc_duplex_bam/biometrics_extract_pickle
    type: File
    'sbg:x': 126.59205627441406
    'sbg:y': -113.84050750732422
steps:
  - id: qc_collapsed_bam
    in:
      - id: reference
        source: reference
      - id: pool_b_target_intervals
        source: pool_b_target_intervals
      - id: pool_a_target_intervals
        source: pool_a_target_intervals
      - id: collapsed_bam
        source:
          - collapsed_bam
      - id: group_reads_by_umi_bam
        source:
          - group_reads_by_umi_bam
      - id: pool_a_bait_intervals
        source: pool_a_bait_intervals
      - id: pool_b_bait_intervals
        source: pool_b_bait_intervals
      - id: json
        source: biometrics_json
      - id: plot
        source: biometrics_plot
      - id: minor_threshold
        source: collapsed_biometrics_minor_threshold
      - id: coverage_threshold
        source: collapsed_biometrics_coverage_threshold
      - id: hsmetrics_minimum_mapping_quality
        source: hsmetrics_minimum_mapping_quality
      - id: hsmetrics_minimum_base_quality
        source: hsmetrics_minimum_base_quality
      - id: hsmetrics_coverage_cap
        source: hsmetrics_coverage_cap
      - id: vcf_file
        source: vcf_file
      - id: sample_sex
        source:
          - sample_sex
      - id: sample_name
        linkMerge: merge_flattened
        source:
          - sample_name
      - id: sample_group
        linkMerge: merge_flattened
        source:
          - sample_group
    out:
      - id: fgbio_collect_duplex_seq_metrics_duplex_family_size_pool_a
      - id: fgbio_collect_duplex_seq_metrics_duplex_qc_pool_a
      - id: fgbio_collect_duplex_seq_metrics_duplex_pool_a
      - id: fgbio_collect_duplex_seq_metrics_duplex_yield_metrics_pool_a
      - id: fgbio_collect_duplex_seq_metrics_family_size_pool_a
      - id: fgbio_collect_duplex_seq_metrics_umi_counts_pool_a
      - id: fgbio_collect_duplex_seq_metrics_duplex_family_size_pool_b
      - id: fgbio_collect_duplex_seq_metrics_duplex_qc_pool_b
      - id: fgbio_collect_duplex_seq_metrics_duplex_umi_counts_pool_b
      - id: fgbio_collect_duplex_seq_metrics_duplex_yield_metrics_pool_b
      - id: fgbio_collect_duplex_seq_metrics_family_size_pool_b
      - id: fgbio_collect_duplex_seq_metrics_umi_counts_pool_b
      - id: biometrics_minor_csv
      - id: biometrics_minor_json
      - id: biometrics_minor_plot
      - id: biometrics_minor_sites_plot
      - id: biometrics_sexmismatch_json
      - id: biometrics_sexmismatch_csv
      - id: gatk_collect_insert_size_metrics_txt_pool_b
      - id: gatk_collect_insert_size_metrics_histogram_pdf_pool_b
      - id: gatk_collect_hs_metrics_txt_pool_b
      - id: gatk_collect_hs_metrics_per_target_coverage_txt_pool_b
      - id: gatk_collect_hs_metrics_per_base_coverage_txt_pool_b
      - id: gatk_collect_alignment_summary_metrics_txt_pool_b
      - id: gatk_collect_insert_size_metrics_txt_pool_a
      - id: gatk_collect_insert_size_metrics_histogram_pdf_pool_a
      - id: gatk_collect_hs_metrics_txt_pool_a
      - id: gatk_collect_hs_metrics_per_target_coverage_txt_pool_a
      - id: gatk_collect_hs_metrics_per_base_coverage_txt_pool_a
      - id: gatk_collect_alignment_summary_metrics_txt_pool_a
      - id: biometrics_extract_pickle
      - id: biometrics_major_plot
      - id: biometrics_major_json
      - id: biometrics_major_csv
    run: ../cwl_subworkflows/qc_collapsed_bam/qc_collapsed_bam.cwl
    label: qc_collapsed_bam
    'sbg:x': -118.83145141601562
    'sbg:y': 289.04119873046875
  - id: qc_uncollapsed_bam
    in:
      - id: reference
        source: reference
      - id: uncollapsed_bam_base_recal
        source:
          - uncollapsed_bam_base_recal
      - id: pool_b_target_intervals
        source: pool_b_target_intervals
      - id: pool_b_bait_intervals
        source: pool_b_bait_intervals
      - id: pool_a_bait_intervals
        source: pool_a_bait_intervals
      - id: pool_a_target_intervals
        source: pool_a_target_intervals
      - id: hsmetrics_minimum_mapping_quality
        source: hsmetrics_minimum_mapping_quality
      - id: hsmetrics_minimum_base_quality
        source: hsmetrics_minimum_base_quality
      - id: hsmetrics_coverage_cap
        source: hsmetrics_coverage_cap
      - id: uncollapsed_bam
        source:
          - uncollapsed_bam
    out:
      - id: gatk_collect_alignment_summary_metrics_txt_pool_b
      - id: gatk_collect_hs_metrics_per_base_coverage_txt_pool_b
      - id: gatk_collect_hs_metrics_per_target_coverage_txt_pool_b
      - id: gatk_collect_hs_metrics_txt_pool_b
      - id: gatk_collect_insert_size_metrics_histogram_pdf_pool_b
      - id: gatk_collect_insert_size_metrics_txt_pool_b
      - id: gatk_collect_alignment_summary_metrics_txt_pool_a
      - id: gatk_collect_hs_metrics_per_base_coverage_txt_pool_a
      - id: gatk_collect_hs_metrics_per_target_coverage_txt_pool_a
      - id: gatk_collect_hs_metrics_txt_pool_a
      - id: gatk_collect_insert_size_metrics_histogram_pdf_pool_a
      - id: gatk_collect_insert_size_metrics_txt_pool_a
      - id: gatk_mean_quality_by_cycle_output
      - id: gatk_mean_quality_by_cycle_chart_output
      - id: gatk_mean_quality_by_cycle_output_base_recal
      - id: gatk_mean_quality_by_cycle_chart_output_base_recal
    run: ../cwl_subworkflows/qc_uncollapsed_bam/qc_uncollapsed_bam.cwl
    label: qc_uncollapsed_bam
    'sbg:x': -77.46428680419922
    'sbg:y': 1069.7781982421875
  - id: qc_duplex_bam
    in:
      - id: reference
        source: reference
      - id: duplex_bam
        source:
          - duplex_bam
      - id: pool_a_target_intervals
        source: pool_a_target_intervals
      - id: pool_a_bait_intervals
        source: pool_a_bait_intervals
      - id: pool_b_target_intervals
        source: pool_b_target_intervals
      - id: pool_b_bait_intervals
        source: pool_b_bait_intervals
      - id: noise_sites_bed
        source: noise_sites_bed
      - id: sample_name
        source:
          - sample_name
      - id: plot
        source: biometrics_plot
      - id: minor_threshold
        source: duplex_biometrics_minor_threshold
      - id: json
        source: biometrics_json
      - id: sequence_qc_min_basq
        source: sequence_qc_min_basq
      - id: sequence_qc_min_mapq
        source: sequence_qc_min_mapq
      - id: sequence_qc_threshold
        source: sequence_qc_threshold
      - id: sequence_qc_truncate
        source: sequence_qc_truncate
      - id: hsmetrics_minimum_mapping_quality
        source: hsmetrics_minimum_mapping_quality
      - id: hsmetrics_minimum_base_quality
        source: hsmetrics_minimum_base_quality
      - id: hsmetrics_coverage_cap
        source: hsmetrics_coverage_cap
      - id: vcf_file
        source: vcf_file
      - id: sample_sex
        source:
          - sample_sex
      - id: sample_group
        linkMerge: merge_flattened
        source:
          - sample_group
      - id: major_threshold
        source: duplex_biometrics_major_threshold
    out:
      - id: biometrics_minor_csv
      - id: biometrics_minor_plot
      - id: biometrics_minor_json
      - id: biometrics_minor_sites_plot
      - id: sequence_qc_noise_positions
      - id: sequence_qc_noise_n
      - id: sequence_qc_noise_del
      - id: sequence_qc_noise_acgt
      - id: sequence_qc_figures
      - id: gatk_collect_alignment_summary_metrics_txt_pool_b
      - id: gatk_collect_hs_metrics_per_base_coverage_txt_pool_b
      - id: gatk_collect_hs_metrics_per_target_coverage_txt_pool_b
      - id: gatk_collect_hs_metrics_txt_pool_b
      - id: gatk_collect_insert_size_metrics_histogram_pdf_pool_b
      - id: gatk_collect_insert_size_metrics_txt_pool_b
      - id: gatk_collect_alignment_summary_metrics_txt_pool_a
      - id: gatk_collect_hs_metrics_per_base_coverage_txt_pool_a
      - id: gatk_collect_hs_metrics_per_target_coverage_txt_pool_a
      - id: gatk_collect_hs_metrics_txt_pool_a
      - id: gatk_collect_insert_size_metrics_histogram_pdf_pool_a
      - id: gatk_collect_insert_size_metrics_txt_pool_a
      - id: sequence_qc_pileup
      - id: sequence_qc_noise_by_substitution
      - id: biometrics_extract_pickle
      - id: biometrics_major_plot
      - id: biometrics_major_json
      - id: biometrics_major_csv
    run: ../cwl_subworkflows/qc_duplex_bam/qc_duplex_bam.cwl
    label: qc_duplex_bam
    'sbg:x': -111.68614196777344
    'sbg:y': -453
  - id: simplex_bam_pool_a
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_simplex_bam/gatk_collect_insert_size_metrics_txt_pool_a
          - qc_simplex_bam/gatk_collect_insert_size_metrics_histogram_pdf_pool_a
          - qc_simplex_bam/gatk_collect_hs_metrics_txt_pool_a
          - >-
            qc_simplex_bam/gatk_collect_hs_metrics_per_target_coverage_txt_pool_a
          - qc_simplex_bam/gatk_collect_hs_metrics_per_base_coverage_txt_pool_a
          - qc_simplex_bam/gatk_collect_alignment_summary_metrics_txt_pool_a
      - id: output_directory_name
        default: simplex_bam_pool_a
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: simplex_bam_pool_a
    'sbg:x': 439
    'sbg:y': -1233.542236328125
  - id: simplex_bam_pool_b
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_simplex_bam/gatk_collect_insert_size_metrics_txt_pool_b
          - qc_simplex_bam/gatk_collect_insert_size_metrics_histogram_pdf_pool_b
          - qc_simplex_bam/gatk_collect_hs_metrics_txt_pool_b
          - >-
            qc_simplex_bam/gatk_collect_hs_metrics_per_target_coverage_txt_pool_b
          - qc_simplex_bam/gatk_collect_hs_metrics_per_base_coverage_txt_pool_b
          - qc_simplex_bam/gatk_collect_alignment_summary_metrics_txt_pool_b
      - id: output_directory_name
        default: simplex_bam_pool_b
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: simplex_bam_pool_b
    'sbg:x': 427.52984619140625
    'sbg:y': -1054.0721435546875
  - id: uncollapsed_bam_stats_pool_b
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_uncollapsed_bam/gatk_collect_alignment_summary_metrics_txt_pool_b
          - >-
            qc_uncollapsed_bam/gatk_collect_hs_metrics_per_base_coverage_txt_pool_b
          - >-
            qc_uncollapsed_bam/gatk_collect_hs_metrics_per_target_coverage_txt_pool_b
          - qc_uncollapsed_bam/gatk_collect_hs_metrics_txt_pool_b
          - >-
            qc_uncollapsed_bam/gatk_collect_insert_size_metrics_histogram_pdf_pool_b
          - qc_uncollapsed_bam/gatk_collect_insert_size_metrics_txt_pool_b
      - id: output_directory_name
        default: uncollapsed_bam_stats_pool_b
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: uncollapsed_bam_stats_pool_b
    'sbg:x': 395.6892395019531
    'sbg:y': 1122.4478759765625
  - id: uncollapsed_bam_stats_pool_a
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_uncollapsed_bam/gatk_collect_alignment_summary_metrics_txt_pool_a
          - >-
            qc_uncollapsed_bam/gatk_collect_hs_metrics_per_base_coverage_txt_pool_a
          - >-
            qc_uncollapsed_bam/gatk_collect_hs_metrics_per_target_coverage_txt_pool_a
          - qc_uncollapsed_bam/gatk_collect_hs_metrics_txt_pool_a
          - >-
            qc_uncollapsed_bam/gatk_collect_insert_size_metrics_histogram_pdf_pool_a
          - qc_uncollapsed_bam/gatk_collect_insert_size_metrics_txt_pool_a
      - id: output_directory_name
        default: uncollapsed_bam_stats_pool_a
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: uncollapsed_bam_stats_pool_a
    'sbg:x': 402.8958740234375
    'sbg:y': 1272.7586669921875
  - id: gatk_mean_quality_by_cycle
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_uncollapsed_bam/gatk_mean_quality_by_cycle_chart_output
          - qc_uncollapsed_bam/gatk_mean_quality_by_cycle_output
      - id: output_directory_name
        default: gatk_mean_quality_by_cycle
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: gatk_mean_quality_by_cycle
    'sbg:x': 403.6803283691406
    'sbg:y': 975.385986328125
  - id: gatk_mean_quality_by_cycle_recal
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - >-
            qc_uncollapsed_bam/gatk_mean_quality_by_cycle_chart_output_base_recal
          - qc_uncollapsed_bam/gatk_mean_quality_by_cycle_output_base_recal
      - id: output_directory_name
        default: gatk_mean_quality_by_cycle_recal
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: gatk_mean_quality_by_cycle_recal
    'sbg:x': 403.2690124511719
    'sbg:y': 829.990234375
  - id: collapsed_bam_stats_pool_a
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_collapsed_bam/gatk_collect_insert_size_metrics_txt_pool_a
          - >-
            qc_collapsed_bam/gatk_collect_insert_size_metrics_histogram_pdf_pool_a
          - qc_collapsed_bam/gatk_collect_hs_metrics_txt_pool_a
          - >-
            qc_collapsed_bam/gatk_collect_hs_metrics_per_target_coverage_txt_pool_a
          - >-
            qc_collapsed_bam/gatk_collect_hs_metrics_per_base_coverage_txt_pool_a
          - qc_collapsed_bam/gatk_collect_alignment_summary_metrics_txt_pool_a
      - id: output_directory_name
        default: collapsed_bam_stats_pool_a
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: collapsed_bam_stats_pool_a
    'sbg:x': 693.8934326171875
    'sbg:y': -116.73040008544922
  - id: collapsed_bam_stats_pool_b
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_collapsed_bam/gatk_collect_insert_size_metrics_txt_pool_b
          - >-
            qc_collapsed_bam/gatk_collect_insert_size_metrics_histogram_pdf_pool_b
          - qc_collapsed_bam/gatk_collect_hs_metrics_txt_pool_b
          - >-
            qc_collapsed_bam/gatk_collect_hs_metrics_per_target_coverage_txt_pool_b
          - >-
            qc_collapsed_bam/gatk_collect_hs_metrics_per_base_coverage_txt_pool_b
          - qc_collapsed_bam/gatk_collect_alignment_summary_metrics_txt_pool_b
      - id: output_directory_name
        default: collapsed_bam_stats_pool_b
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: collapsed_bam_stats_pool_b
    'sbg:x': 693
    'sbg:y': 13.673991203308105
  - id: collapsed_bam_duplex_metrics_pool_a
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_collapsed_bam/fgbio_collect_duplex_seq_metrics_umi_counts_pool_a
          - qc_collapsed_bam/fgbio_collect_duplex_seq_metrics_family_size_pool_a
          - >-
            qc_collapsed_bam/fgbio_collect_duplex_seq_metrics_duplex_yield_metrics_pool_a
          - qc_collapsed_bam/fgbio_collect_duplex_seq_metrics_duplex_qc_pool_a
          - qc_collapsed_bam/fgbio_collect_duplex_seq_metrics_duplex_pool_a
          - >-
            qc_collapsed_bam/fgbio_collect_duplex_seq_metrics_duplex_family_size_pool_a
      - id: output_directory_name
        default: collapsed_bam_duplex_metrics_pool_a
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: collapsed_bam_duplex_metrics_pool_a
    'sbg:x': 690.4910278320312
    'sbg:y': 137.0360565185547
  - id: collapsed_bam_duplex_metrics_pool_b
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_collapsed_bam/fgbio_collect_duplex_seq_metrics_umi_counts_pool_b
          - qc_collapsed_bam/fgbio_collect_duplex_seq_metrics_family_size_pool_b
          - >-
            qc_collapsed_bam/fgbio_collect_duplex_seq_metrics_duplex_yield_metrics_pool_b
          - >-
            qc_collapsed_bam/fgbio_collect_duplex_seq_metrics_duplex_umi_counts_pool_b
          - qc_collapsed_bam/fgbio_collect_duplex_seq_metrics_duplex_qc_pool_b
          - >-
            qc_collapsed_bam/fgbio_collect_duplex_seq_metrics_duplex_family_size_pool_b
      - id: output_directory_name
        default: collapsed_bam_duplex_metrics_pool_b
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: collapsed_bam_duplex_metrics_pool_b
    'sbg:x': 686.7992553710938
    'sbg:y': 265.29779052734375
  - id: collapsed_bam_biometrics
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_collapsed_bam/biometrics_sexmismatch_json
          - qc_collapsed_bam/biometrics_sexmismatch_csv
          - qc_collapsed_bam/biometrics_minor_sites_plot
          - qc_collapsed_bam/biometrics_minor_plot
          - qc_collapsed_bam/biometrics_minor_json
          - qc_collapsed_bam/biometrics_minor_csv
          - qc_collapsed_bam/biometrics_major_plot
          - qc_collapsed_bam/biometrics_major_json
          - qc_collapsed_bam/biometrics_major_csv
          - qc_collapsed_bam/biometrics_extract_pickle
      - id: output_directory_name
        default: collapsed_bam_biometrics
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: collapsed_bam_biometrics
    'sbg:x': 682.2883911132812
    'sbg:y': 410.1722412109375
  - id: duplex_bam_sequence_qc
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_duplex_bam/sequence_qc_noise_positions
          - qc_duplex_bam/sequence_qc_noise_n
          - qc_duplex_bam/sequence_qc_noise_del
          - qc_duplex_bam/sequence_qc_noise_acgt
          - qc_duplex_bam/sequence_qc_figures
          - qc_duplex_bam/sequence_qc_pileup
      - id: output_directory_name
        default: duplex_bam_sequence_qc
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: duplex_bam_sequence_qc
    'sbg:x': 453.6691589355469
    'sbg:y': -790.8209228515625
  - id: duplex_bam_stats_pool_a
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_duplex_bam/gatk_collect_insert_size_metrics_txt_pool_a
          - qc_duplex_bam/gatk_collect_insert_size_metrics_histogram_pdf_pool_a
          - qc_duplex_bam/gatk_collect_hs_metrics_txt_pool_a
          - qc_duplex_bam/gatk_collect_hs_metrics_per_target_coverage_txt_pool_a
          - qc_duplex_bam/gatk_collect_hs_metrics_per_base_coverage_txt_pool_a
          - qc_duplex_bam/gatk_collect_alignment_summary_metrics_txt_pool_a
      - id: output_directory_name
        default: duplex_bam_stats_pool_a
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: duplex_bam_stats_pool_a
    'sbg:x': 530.0328979492188
    'sbg:y': -595.6776123046875
  - id: duplex_bam_stats_pool_b
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_duplex_bam/gatk_collect_insert_size_metrics_txt_pool_b
          - qc_duplex_bam/gatk_collect_insert_size_metrics_histogram_pdf_pool_b
          - qc_duplex_bam/gatk_collect_hs_metrics_txt_pool_b
          - qc_duplex_bam/gatk_collect_hs_metrics_per_target_coverage_txt_pool_b
          - qc_duplex_bam/gatk_collect_hs_metrics_per_base_coverage_txt_pool_b
          - qc_duplex_bam/gatk_collect_alignment_summary_metrics_txt_pool_b
      - id: output_directory_name
        default: duplex_bam_stats_pool_b
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: duplex_bam_stats_pool_b
    'sbg:x': 530.8358764648438
    'sbg:y': -479.8985290527344
  - id: duplex_bam_biometrics
    in:
      - id: files
        linkMerge: merge_flattened
        source:
          - qc_duplex_bam/biometrics_major_csv
          - qc_duplex_bam/biometrics_major_json
          - qc_duplex_bam/biometrics_major_plot
          - qc_duplex_bam/biometrics_minor_csv
          - qc_duplex_bam/biometrics_minor_json
          - qc_duplex_bam/biometrics_minor_plot
          - qc_duplex_bam/biometrics_minor_sites_plot
          - qc_duplex_bam/biometrics_extract_pickle
      - id: output_directory_name
        default: duplex_bam_biometrics
    out:
      - id: directory
    run: ../cwl-commandlinetools/expression_tools/put_in_dir.cwl
    label: duplex_bam_biometrics
    'sbg:x': 526.955322265625
    'sbg:y': -361.4269104003906
  - id: qc_simplex_bam
    in:
      - id: reference
        source: reference
      - id: simplex_bam
        source: simplex_bam
      - id: pool_b_target_intervals
        source: pool_b_target_intervals
      - id: pool_b_bait_intervals
        source: pool_b_bait_intervals
      - id: pool_a_bait_intervals
        source: pool_a_bait_intervals
      - id: pool_a_target_intervals
        source: pool_a_target_intervals
      - id: hsmetrics_minimum_mapping_quality
        source: hsmetrics_minimum_mapping_quality
      - id: hsmetrics_minimum_base_quality
        source: hsmetrics_minimum_base_quality
      - id: hsmetrics_coverage_cap
        source: hsmetrics_coverage_cap
    out:
      - id: gatk_collect_alignment_summary_metrics_txt_pool_b
      - id: gatk_collect_hs_metrics_per_base_coverage_txt_pool_b
      - id: gatk_collect_hs_metrics_per_target_coverage_txt_pool_b
      - id: gatk_collect_hs_metrics_txt_pool_b
      - id: gatk_collect_insert_size_metrics_histogram_pdf_pool_b
      - id: gatk_collect_insert_size_metrics_txt_pool_b
      - id: gatk_collect_alignment_summary_metrics_txt_pool_a
      - id: gatk_collect_hs_metrics_per_base_coverage_txt_pool_a
      - id: gatk_collect_hs_metrics_per_target_coverage_txt_pool_a
      - id: gatk_collect_hs_metrics_txt_pool_a
      - id: gatk_collect_insert_size_metrics_histogram_pdf_pool_a
      - id: gatk_collect_insert_size_metrics_txt_pool_a
    run: ../cwl_subworkflows/qc_simplex_bam/qc_simplex_bam.cwl
    label: qc_simplex_bam
    'sbg:x': -129.99029541015625
    'sbg:y': -1110.5147705078125
requirements:
  - class: SubworkflowFeatureRequirement
  - class: MultipleInputFeatureRequirement
$schemas:
  - 'http://schema.org/version/latest/schemaorg-current-http.rdf'
's:author':
  - class: 's:Person'
    's:email': 'mailto:murphyc4@mskcc.org'
    's:identifier': ''
    's:name': Charlie Murphy
's:citation': ''
's:codeRepository': 'https://github.com/msk-access/cwl_subworkflows'
's:contributor':
  - class: 's:Person'
    's:email': 'mailto:murphyc4@mskcc.org'
    's:name': Charlie Murphy
's:dateCreated': '2021-05-19'
's:license': 'https://spdx.org/licenses/Apache-2.0'
