class: Workflow
cwlVersion: v1.0
id: qc_collapsed_bam
label: qc_collapsed_bam
$namespaces:
  sbg: 'https://www.sevenbridges.com/'
inputs:
  - id: reference
    type: File
    secondaryFiles:
      - ^.fasta.fai
      - ^.dict
    'sbg:x': -1379.3106689453125
    'sbg:y': -9
  - id: pool_b_target_intervals
    type: File
    label: pool_b_target_intervals
    'sbg:x': -1380.0771484375
    'sbg:y': -176.342529296875
  - id: pool_a_target_intervals
    type: File
    label: pool_a_target_intervals
    'sbg:x': -1407.2725830078125
    'sbg:y': -725.6703491210938
  - id: biometrics_vcf_file
    type: File
    doc: VCF file containing the SNPs to be queried.
    'sbg:x': -1376.3106689453125
    'sbg:y': 160.8839111328125
  - id: collapsed_bam
    type:
      - File
      - type: array
        items: File
    label: collapsed_bam
    secondaryFiles:
      - ^.bai
    'sbg:x': -899.7366333007812
    'sbg:y': 462.836181640625
  - id: sample_type
    type:
      - 'null'
      - string
      - type: array
        items: string
    doc: 'Sample types: Normal or Tumor.'
    'sbg:x': -900.1541137695312
    'sbg:y': 613.905517578125
  - id: sample_sex
    type:
      - 'null'
      - string
      - type: array
        items: string
    doc: Expected sample sex (i.e. M or F).
    'sbg:x': -909.779296875
    'sbg:y': 779.0851440429688
  - id: sample_name
    type:
      - 'null'
      - string
      - type: array
        items: string
    doc: >-
      Sample name. If not specified, sample name is automatically figured out
      from the BAM file.
    'sbg:x': -913.1387939453125
    'sbg:y': 910.29150390625
  - id: sample_group
    type:
      - 'null'
      - string
      - type: array
        items: string
    doc: The sample group (e.g. the sample patient ID).
    'sbg:x': -907.56005859375
    'sbg:y': 1060.3988037109375
  - id: group_reads_by_umi_bam
    type:
      - File
      - type: array
        items: File
    label: group_reads_by_umi_bam
    doc: Input BAM file generated by GroupReadByUmi.
    'sbg:x': -911.3615112304688
    'sbg:y': 324.525634765625
  - id: pool_a_bait_intervals
    type: File?
    label: pool_a_bait_intervals
    doc: 'Optional set of intervals over which to restrict analysis. [Optional].'
    'sbg:x': -1419.413818359375
    'sbg:y': -885.5172119140625
  - id: pool_b_bait_intervals
    type: File?
    label: pool_b_bait_intervals
    doc: 'Optional set of intervals over which to restrict analysis. [Optional].'
    'sbg:x': -1393.6268310546875
    'sbg:y': -352.0533142089844
  - id: biometrics_bed_file
    type: File?
    doc: BED file containing the intervals to be queried.
    'sbg:x': -1384.4722900390625
    'sbg:y': 347.15325927734375
  - id: json
    type: boolean?
    doc: Also output data in JSON format.
    'sbg:x': -50.14529037475586
    'sbg:y': 668.078369140625
  - id: plot
    type: boolean?
    doc: Also output plots of the data.
    'sbg:x': -37.99789047241211
    'sbg:y': 828.6326904296875
  - id: major_threshold
    type: float?
    doc: Major contamination threshold for bad sample.
    'sbg:x': -37.99789047241211
    'sbg:y': 986.5403442382812
  - id: minor_threshold
    type: float?
    doc: Minor contamination threshold for bad sample.
    'sbg:x': -20.255456924438477
    'sbg:y': 1140.8995361328125
  - id: coverage_threshold
    type: int?
    doc: Samples with Y chromosome above this value will be considered male.
    'sbg:x': -28.91999626159668
    'sbg:y': 1299.155029296875
  - id: min_mapping_quality
    type: int?
    doc: Minimum mapping quality of reads to be used for pileup.
    'sbg:x': -316.75909423828125
    'sbg:y': 658.3980102539062
  - id: min_homozygous_thresh
    type: float?
    doc: Minimum threshold to define homozygous.
    'sbg:x': -310.90899658203125
    'sbg:y': 805.6259155273438
  - id: min_coverage
    type: int?
    doc: Minimum coverage to count a site.
    'sbg:x': -304.0838623046875
    'sbg:y': 946.0287475585938
  - id: min_base_quality
    type: int?
    doc: Minimum base quality of reads to be used for pileup.
    'sbg:x': -313.83404541015625
    'sbg:y': 1075.706298828125
  - id: hsmetrics_minimum_mapping_quality
    type: int?
    'sbg:x': -1416.0538330078125
    'sbg:y': -1003.3903198242188
  - id: hsmetrics_minimum_base_quality
    type: int?
    'sbg:x': -1417.81689453125
    'sbg:y': -1132.09375
  - id: hsmetrics_coverage_cap
    type: int?
    'sbg:x': -1414.290771484375
    'sbg:y': -1262.5513916015625
  - id: prefix
    type: string?
    'sbg:x': -26.909893035888672
    'sbg:y': 1447.054931640625
outputs:
  - id: biometrics_extract_pickle
    outputSource:
      - biometrics_extract/biometrics_extract_pickle
    type:
      - File
      - type: array
        items: File
    'sbg:x': 1073.549560546875
    'sbg:y': -58.49618148803711
  - id: fgbio_collect_duplex_seq_metrics_duplex_family_size_pool_a
    outputSource:
      - >-
        fgbio_collect_duplex_seq_metrics_1_2_0/fgbio_collect_duplex_seq_metrics_duplex_family_size
    type:
      - File
      - type: array
        items: File
    label: fgbio_collect_duplex_seq_metrics_duplex_family_size_pool_a
    'sbg:x': 516.4937744140625
    'sbg:y': -1038.1038818359375
  - id: fgbio_collect_duplex_seq_metrics_duplex_qc_pool_a
    outputSource:
      - >-
        fgbio_collect_duplex_seq_metrics_1_2_0/fgbio_collect_duplex_seq_metrics_duplex_qc
    type:
      - 'null'
      - File
      - type: array
        items: File
    label: fgbio_collect_duplex_seq_metrics_duplex_qc_pool_a
    'sbg:x': 507.3194885253906
    'sbg:y': -1158.8990478515625
  - id: fgbio_collect_duplex_seq_metrics_duplex_pool_a
    outputSource:
      - >-
        fgbio_collect_duplex_seq_metrics_1_2_0/fgbio_collect_duplex_seq_metrics_duplex_umi_counts
    type:
      - 'null'
      - File
      - type: array
        items: File
    label: fgbio_collect_duplex_seq_metrics_duplex_pool_a
    'sbg:x': 510.3775939941406
    'sbg:y': -1284.28125
  - id: fgbio_collect_duplex_seq_metrics_duplex_yield_metrics_pool_a
    outputSource:
      - >-
        fgbio_collect_duplex_seq_metrics_1_2_0/fgbio_collect_duplex_seq_metrics_duplex_yield_metrics
    type:
      - File
      - type: array
        items: File
    label: fgbio_collect_duplex_seq_metrics_duplex_yield_metrics_pool_a
    'sbg:x': 514.9647216796875
    'sbg:y': -1418.837890625
  - id: fgbio_collect_duplex_seq_metrics_family_size_pool_a
    outputSource:
      - >-
        fgbio_collect_duplex_seq_metrics_1_2_0/fgbio_collect_duplex_seq_metrics_family_size
    type:
      - File
      - type: array
        items: File
    label: fgbio_collect_duplex_seq_metrics_family_size_pool_a
    'sbg:x': 516.4937744140625
    'sbg:y': -1547.2781982421875
  - id: fgbio_collect_duplex_seq_metrics_umi_counts_pool_a
    outputSource:
      - >-
        fgbio_collect_duplex_seq_metrics_1_2_0/fgbio_collect_duplex_seq_metrics_umi_counts
    type:
      - File
      - type: array
        items: File
    label: fgbio_collect_duplex_seq_metrics_umi_counts_pool_a
    'sbg:x': 516.4937744140625
    'sbg:y': -1683.3638916015625
  - id: fgbio_collect_duplex_seq_metrics_duplex_family_size_pool_b
    outputSource:
      - >-
        fgbio_collect_duplex_seq_metrics_1_2_1/fgbio_collect_duplex_seq_metrics_duplex_family_size
    type:
      - File
      - type: array
        items: File
    label: fgbio_collect_duplex_seq_metrics_duplex_family_size_pool_b
    'sbg:x': 510.3775939941406
    'sbg:y': -1842.38525390625
  - id: fgbio_collect_duplex_seq_metrics_duplex_qc_pool_b
    outputSource:
      - >-
        fgbio_collect_duplex_seq_metrics_1_2_1/fgbio_collect_duplex_seq_metrics_duplex_qc
    type:
      - 'null'
      - File
      - type: array
        items: File
    label: fgbio_collect_duplex_seq_metrics_duplex_qc_pool_b
    'sbg:x': 508.8485412597656
    'sbg:y': -1998.3485107421875
  - id: fgbio_collect_duplex_seq_metrics_duplex_umi_counts_pool_b
    outputSource:
      - >-
        fgbio_collect_duplex_seq_metrics_1_2_1/fgbio_collect_duplex_seq_metrics_duplex_umi_counts
    type:
      - 'null'
      - File
      - type: array
        items: File
    label: fgbio_collect_duplex_seq_metrics_duplex_umi_counts_pool_b
    'sbg:x': 499.6742248535156
    'sbg:y': -2158.89892578125
  - id: fgbio_collect_duplex_seq_metrics_duplex_yield_metrics_pool_b
    outputSource:
      - >-
        fgbio_collect_duplex_seq_metrics_1_2_1/fgbio_collect_duplex_seq_metrics_duplex_yield_metrics
    type:
      - File
      - type: array
        items: File
    label: fgbio_collect_duplex_seq_metrics_duplex_yield_metrics_pool_b
    'sbg:x': 495.0870666503906
    'sbg:y': -2319.449462890625
  - id: fgbio_collect_duplex_seq_metrics_family_size_pool_b
    outputSource:
      - >-
        fgbio_collect_duplex_seq_metrics_1_2_1/fgbio_collect_duplex_seq_metrics_family_size
    type:
      - File
      - type: array
        items: File
    label: fgbio_collect_duplex_seq_metrics_family_size_pool_b
    'sbg:x': 502.7323303222656
    'sbg:y': -2457.064208984375
  - id: fgbio_collect_duplex_seq_metrics_umi_counts_pool_b
    outputSource:
      - >-
        fgbio_collect_duplex_seq_metrics_1_2_1/fgbio_collect_duplex_seq_metrics_umi_counts
    type:
      - File
      - type: array
        items: File
    label: fgbio_collect_duplex_seq_metrics_umi_counts_pool_b
    'sbg:x': 498.1451721191406
    'sbg:y': -2605.382080078125
  - id: biometrics_minor_csv
    outputSource:
      - biometrics_minor/biometrics_minor_csv
    type:
      - File
      - type: array
        items: File
    'sbg:x': 1108.409912109375
    'sbg:y': 879.7926025390625
  - id: biometrics_minor_json
    outputSource:
      - biometrics_minor/biometrics_minor_json
    type:
      - 'null'
      - File
      - type: array
        items: File
    'sbg:x': 1099.409912109375
    'sbg:y': 734.1456909179688
  - id: biometrics_minor_plot
    outputSource:
      - biometrics_minor/biometrics_minor_plot
    type:
      - 'null'
      - File
      - type: array
        items: File
    'sbg:x': 1092.851806640625
    'sbg:y': 577.2938232421875
  - id: biometrics_minor_sites_plot
    outputSource:
      - biometrics_minor/biometrics_minor_sites_plot
    type:
      - 'null'
      - File
      - type: array
        items: File
    'sbg:x': 1085.1456298828125
    'sbg:y': 460.587646484375
  - id: biometrics_major_csv
    outputSource:
      - biometrics_major/biometrics_major_csv
    type:
      - File
      - type: array
        items: File
    'sbg:x': 1085.49560546875
    'sbg:y': 332.7539367675781
  - id: biometrics_major_json
    outputSource:
      - biometrics_major/biometrics_major_json
    type:
      - 'null'
      - File
      - type: array
        items: File
    'sbg:x': 1089.466064453125
    'sbg:y': 211.00634765625
  - id: biometrics_major_plot
    outputSource:
      - biometrics_major/biometrics_major_plot
    type:
      - 'null'
      - File
      - type: array
        items: File
    'sbg:x': 1077.554931640625
    'sbg:y': 82.63099670410156
  - id: biometrics_sexmismatch_json
    outputSource:
      - biometrics_sexmismatch/biometrics_sexmismatch_json
    type:
      - 'null'
      - File
      - type: array
        items: File
    'sbg:x': 1105.6673583984375
    'sbg:y': 1000.427490234375
  - id: biometrics_sexmismatch_csv
    outputSource:
      - biometrics_sexmismatch/biometrics_sexmismatch_csv
    type:
      - File
      - type: array
        items: File
    'sbg:x': 1098.1990966796875
    'sbg:y': 1146.0594482421875
  - id: gatk_collect_insert_size_metrics_txt_pool_b
    outputSource:
      - bam_qc_stats_pool_b/gatk_collect_insert_size_metrics_txt
    type:
      - File
      - type: array
        items: File
    label: gatk_collect_insert_size_metrics_txt_pool_b
    'sbg:x': 466.4290771484375
    'sbg:y': -441.6470642089844
  - id: gatk_collect_insert_size_metrics_histogram_pdf_pool_b
    outputSource:
      - bam_qc_stats_pool_b/gatk_collect_insert_size_metrics_histogram_pdf
    type:
      - File
      - type: array
        items: File
    label: gatk_collect_insert_size_metrics_histogram_pdf_pool_b
    'sbg:x': 469.1591796875
    'sbg:y': -308.7266540527344
  - id: gatk_collect_hs_metrics_txt_pool_b
    outputSource:
      - bam_qc_stats_pool_b/gatk_collect_hs_metrics_txt
    type:
      - File
      - type: array
        items: File
    label: gatk_collect_hs_metrics_txt_pool_b
    'sbg:x': 462.23876953125
    'sbg:y': -168.50518798828125
  - id: gatk_collect_hs_metrics_per_target_coverage_txt_pool_b
    outputSource:
      - bam_qc_stats_pool_b/gatk_collect_hs_metrics_per_target_coverage_txt
    type:
      - File
      - type: array
        items: File
    label: gatk_collect_hs_metrics_per_target_coverage_txt_pool_b
    'sbg:x': 463.8892822265625
    'sbg:y': -41.584774017333984
  - id: gatk_collect_hs_metrics_per_base_coverage_txt_pool_b
    outputSource:
      - bam_qc_stats_pool_b/gatk_collect_hs_metrics_per_base_coverage_txt
    type:
      - File
      - type: array
        items: File
    label: gatk_collect_hs_metrics_per_base_coverage_txt_pool_b
    'sbg:x': 470
    'sbg:y': 76.14533233642578
  - id: gatk_collect_alignment_summary_metrics_txt_pool_b
    outputSource:
      - bam_qc_stats_pool_b/gatk_collect_alignment_summary_metrics_txt
    type:
      - File
      - type: array
        items: File
    label: gatk_collect_alignment_summary_metrics_txt_pool_b
    'sbg:x': 476.1522521972656
    'sbg:y': 197.31488037109375
  - id: gatk_collect_insert_size_metrics_txt_pool_a
    outputSource:
      - bam_qc_stats_pool_a/gatk_collect_insert_size_metrics_txt
    type:
      - File
      - type: array
        items: File
    label: gatk_collect_insert_size_metrics_txt_pool_a
    'sbg:x': 869.6942749023438
    'sbg:y': -947.464111328125
  - id: gatk_collect_insert_size_metrics_histogram_pdf_pool_a
    outputSource:
      - bam_qc_stats_pool_a/gatk_collect_insert_size_metrics_histogram_pdf
    type:
      - File
      - type: array
        items: File
    label: gatk_collect_insert_size_metrics_histogram_pdf_pool_a
    'sbg:x': 861.0437622070312
    'sbg:y': -829.8170776367188
  - id: gatk_collect_hs_metrics_txt_pool_a
    outputSource:
      - bam_qc_stats_pool_a/gatk_collect_hs_metrics_txt
    type:
      - File
      - type: array
        items: File
    label: gatk_collect_hs_metrics_txt_pool_a
    'sbg:x': 859.3136596679688
    'sbg:y': -681.0281372070312
  - id: gatk_collect_hs_metrics_per_target_coverage_txt_pool_a
    outputSource:
      - bam_qc_stats_pool_a/gatk_collect_hs_metrics_per_target_coverage_txt
    type:
      - File
      - type: array
        items: File
    label: gatk_collect_hs_metrics_per_target_coverage_txt_pool_a
    'sbg:x': 866.2340698242188
    'sbg:y': -556.460693359375
  - id: gatk_collect_hs_metrics_per_base_coverage_txt_pool_a
    outputSource:
      - bam_qc_stats_pool_a/gatk_collect_hs_metrics_per_base_coverage_txt
    type:
      - File
      - type: array
        items: File
    label: gatk_collect_hs_metrics_per_base_coverage_txt_pool_a
    'sbg:x': 866.2340698242188
    'sbg:y': -421.5125732421875
  - id: gatk_collect_alignment_summary_metrics_txt_pool_a
    outputSource:
      - bam_qc_stats_pool_a/gatk_collect_alignment_summary_metrics_txt
    type:
      - File
      - type: array
        items: File
    label: gatk_collect_alignment_summary_metrics_txt_pool_a
    'sbg:x': 859.3136596679688
    'sbg:y': -296.9450988769531
steps:
  - id: bam_qc_stats_pool_b
    in:
      - id: input
        source:
          - collapsed_bam
      - id: target_intervals
        source: pool_b_target_intervals
      - id: bait_intervals
        source: pool_b_bait_intervals
      - id: reference
        source: reference
      - id: hsmetrics_minimum_mapping_quality
        source: hsmetrics_minimum_mapping_quality
      - id: hsmetrics_minimum_base_quality
        source: hsmetrics_minimum_base_quality
      - id: hsmetrics_coverage_cap
        source: hsmetrics_coverage_cap
    out:
      - id: gatk_collect_insert_size_metrics_histogram_pdf
      - id: gatk_collect_insert_size_metrics_txt
      - id: gatk_collect_hs_metrics_txt
      - id: gatk_collect_hs_metrics_per_base_coverage_txt
      - id: gatk_collect_hs_metrics_per_target_coverage_txt
      - id: gatk_collect_alignment_summary_metrics_txt
    run: ../cwl_subworkflows/bam_qc_stats/bam_qc_stats.cwl
    label: bam_qc_stats_pool_b
    'sbg:x': 244.58816528320312
    'sbg:y': -98.25187683105469
  - id: bam_qc_stats_pool_a
    in:
      - id: input
        source:
          - collapsed_bam
      - id: target_intervals
        source: pool_a_target_intervals
      - id: bait_intervals
        source: pool_a_bait_intervals
      - id: reference
        source: reference
      - id: hsmetrics_minimum_mapping_quality
        source: hsmetrics_minimum_mapping_quality
      - id: hsmetrics_minimum_base_quality
        source: hsmetrics_minimum_base_quality
      - id: hsmetrics_coverage_cap
        source: hsmetrics_coverage_cap
    out:
      - id: gatk_collect_insert_size_metrics_histogram_pdf
      - id: gatk_collect_insert_size_metrics_txt
      - id: gatk_collect_hs_metrics_txt
      - id: gatk_collect_hs_metrics_per_base_coverage_txt
      - id: gatk_collect_hs_metrics_per_target_coverage_txt
      - id: gatk_collect_alignment_summary_metrics_txt
    run: ../cwl_subworkflows/bam_qc_stats/bam_qc_stats.cwl
    label: bam_qc_stats_pool_a
    'sbg:x': -79.8152847290039
    'sbg:y': -635.7706909179688
  - id: biometrics_extract
    in:
      - id: sample_bam
        linkMerge: merge_nested
        source:
          - collapsed_bam
      - id: sample_type
        linkMerge: merge_nested
        source:
          - sample_type
      - id: sample_sex
        linkMerge: merge_nested
        source:
          - sample_sex
      - id: sample_group
        linkMerge: merge_nested
        source:
          - sample_group
      - id: sample_name
        linkMerge: merge_nested
        source:
          - sample_name
      - id: fafile
        source: reference
      - id: vcf_file
        source: biometrics_vcf_file
      - id: bed_file
        source: biometrics_bed_file
      - id: min_mapping_quality
        source: min_mapping_quality
      - id: min_base_quality
        source: min_base_quality
      - id: min_coverage
        source: min_coverage
      - id: min_homozygous_thresh
        source: min_homozygous_thresh
    out:
      - id: biometrics_extract_pickle
    run: ../cwl-commandlinetools/biometrics_extract/0.2.9/biometrics_extract.cwl
    'sbg:x': -56.18357467651367
    'sbg:y': 437.74395751953125
  - id: fgbio_collect_duplex_seq_metrics_1_2_0
    in:
      - id: input
        source: group_reads_by_umi_bam
      - id: intervals
        source: pool_a_bait_intervals
    out:
      - id: fgbio_collect_duplex_seq_metrics_family_size
      - id: fgbio_collect_duplex_seq_metrics_duplex_family_size
      - id: fgbio_collect_duplex_seq_metrics_duplex_yield_metrics
      - id: fgbio_collect_duplex_seq_metrics_umi_counts
      - id: fgbio_collect_duplex_seq_metrics_duplex_qc
      - id: fgbio_collect_duplex_seq_metrics_duplex_umi_counts
    run: >-
      ../cwl-commandlinetools/fgbio_collect_duplex_seq_metrics_1.2.0/fgbio_collect_duplex_seq_metrics_1.2.0.cwl
    label: fgbio_collect_duplex_seq_metrics_1.2.0
    'sbg:x': -102.85713958740234
    'sbg:y': -1250.857177734375
  - id: fgbio_collect_duplex_seq_metrics_1_2_1
    in:
      - id: input
        source: group_reads_by_umi_bam
      - id: intervals
        source: pool_b_bait_intervals
    out:
      - id: fgbio_collect_duplex_seq_metrics_family_size
      - id: fgbio_collect_duplex_seq_metrics_duplex_family_size
      - id: fgbio_collect_duplex_seq_metrics_duplex_yield_metrics
      - id: fgbio_collect_duplex_seq_metrics_umi_counts
      - id: fgbio_collect_duplex_seq_metrics_duplex_qc
      - id: fgbio_collect_duplex_seq_metrics_duplex_umi_counts
    run: >-
      ../cwl-commandlinetools/fgbio_collect_duplex_seq_metrics_1.2.0/fgbio_collect_duplex_seq_metrics_1.2.0.cwl
    label: fgbio_collect_duplex_seq_metrics_1.2.0
    'sbg:x': -101.767578125
    'sbg:y': -2137.599365234375
  - id: biometrics_major
    in:
      - id: input
        source:
          - biometrics_extract/biometrics_extract_pickle
      - id: major_threshold
        source: major_threshold
      - id: prefix
        default: collapsed
        source: prefix
      - id: plot
        source: plot
      - id: json
        source: json
    out:
      - id: biometrics_major_csv
      - id: biometrics_major_json
      - id: biometrics_major_plot
    run: ../cwl-commandlinetools/biometrics_major/0.2.9/biometrics_major.cwl
    'sbg:x': 677.335205078125
    'sbg:y': 262.2733154296875
  - id: biometrics_minor
    in:
      - id: input
        source:
          - biometrics_extract/biometrics_extract_pickle
      - id: minor_threshold
        source: minor_threshold
      - id: prefix
        default: collapsed
        source: prefix
      - id: plot
        default: false
        source: plot
      - id: json
        default: true
        source: json
    out:
      - id: biometrics_minor_csv
      - id: biometrics_minor_json
      - id: biometrics_minor_plot
      - id: biometrics_minor_sites_plot
    run: ../cwl-commandlinetools/biometrics_minor/0.2.9/biometrics_minor.cwl
    'sbg:x': 686.5601196289062
    'sbg:y': 571.31689453125
  - id: biometrics_sexmismatch
    in:
      - id: input
        source:
          - biometrics_extract/biometrics_extract_pickle
      - id: coverage_threshold
        source: coverage_threshold
      - id: prefix
        default: collapsed
        source: prefix
      - id: json
        source: json
    out:
      - id: biometrics_sexmismatch_csv
      - id: biometrics_sexmismatch_json
    run: >-
      ../cwl-commandlinetools/biometrics_sexmismatch/0.2.9/biometrics_sexmismatch.cwl
    'sbg:x': 688.3497924804688
    'sbg:y': 1029.9459228515625
requirements:
  - class: SubworkflowFeatureRequirement
